packages/dds/tree/src/feature-libraries/defaultFieldKinds.ts

```diff
@@ -698,6 +816,20 @@ export const optional: FieldKind<OptionalFieldEditor, Multiplicity.Optional> = n
 		editor: optionalFieldEditor,
 
 		intoDelta: (change: OptionalChangeset, deltaFromChild: ToDelta) => {
+			if (change.fieldChange === undefined) {
+				if (change.deletedBy === undefined && change.childChange !== undefined) {
+					return [deltaFromChild(change.childChange)];
+				}
+				return [];
+			}
+
+			assert(change.deletedBy === undefined, "Unhandled");
```

Comment:
What scenario is it that we're not handling here?

Reply:
It looks like we just need to pass `undefined` instead of `change.childChange` into the `deltaForDelete` call if `change.deletedBy !== undefined`.

Reply:
I believe I've updated this block as Alex suggested in the latest commit.